src/sidebars/favourites/geometry-service.js

```diff
@@ -45,6 +45,18 @@
                 otherOrigin.x < this.corner().x &&
                 otherOrigin.y < this.corner().y;
         }
+
+        areaOfIntersection(otherRectangle) {
+            if (this.intersects(otherRectangle)) {
+                var left = Math.max(this.left(), otherRectangle.left());
+                var right = Math.min(this.right(), otherRectangle.right());
+                var top = Math.max(this.top(), otherRectangle.top());
+                var bottom = Math.min(this.bottom(), otherRectangle.bottom());
+                return (right - left) * (bottom - top);
+            } else {
```

Comment:
I'm about to change this, but why the farthest left and closest right as opposed to vice versa?